File: algorithms/2206.divide-array-into-equal-pairs/solution.rs
use std::collections::HashSet;

impl Solution {
    pub fn divide_array(nums: Vec<i32>) -> bool {
        let mut hs: HashSet<i32> = HashSet::new();
        for n in nums {
            if let Some(_) = hs.get(&n) {
                hs.remove(&n);
            } else {
                hs.insert(n);
            }
        }

        hs.len() == 0
    }
}

File: algorithms/2206.divide-array-into-equal-pairs/solution.rb
# @param {Integer[]} nums
# @return {Boolean}
def divide_array(nums)
  set = Set.new

  nums.each do |n|
    if set.include?(n)
      set.delete(n)
    else
      set.add(n)
    end
  end

  set.size == 0
end
